### PR TITLE
add VillageGameSecretWellEvent

### PR DESCRIPTION
**Pull request by HSGamer**

### CODE REVIEW

**Comment by HSGamer** [comment]
Handle more items than just rotten fresh

**Comment by HSGamer** [comment]
Any comment on this ??

File: src/main/java/plugily/projects/villagedefense/events/Events.java [hunk 1]
```diff
@@ -433,39 +435,48 @@ public void onCraft(CBPlayerInteractEvent event) {
...
-        if(e.getItem().getWorld().equals(arena.getStartLocation().getWorld())) {
-          e.getItem().remove();
-          e.getInventory().clear();
-          return;
-        }
+    Arena currentArena = null;
+    for(Arena arena : ArenaRegistry.getArenas()) {
+      if(e.getItem().getWorld().equals(arena.getStartLocation().getWorld())) {
+        currentArena = arena;
+        e.getItem().remove();
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/events/Events.java:446]: The item is removed here, so in the `getNearbyEntities` loop the location of this item will be null or unloaded.

File: src/main/java/plugily/projects/villagedefense/events/Events.java [hunk 2]
```diff
@@ -433,37 +436,42 @@ public void onCraft(CBPlayerInteractEvent event) {
...
+        break;
       }
+    }
+    if (currentArena == null) {
       return;
     }
-    for(Entity entity : Utils.getNearbyEntities(e.getItem().getLocation(), 20)) {
-      if(!(entity instanceof Player)) {
-        continue;
-      }
```
[the remark this answers is not in the document]
Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/events/Events.java:453]: I wonder why there is a loop for entities if the goal is to increase the max health of all players in the arena.

[the remark this answers is not in the document]
Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/events/Events.java:453]: If the goal is to increase the max health based on the players near the well then ok

**Comment by HSGamer** [comment]
Was that enough?

File: src/main/java/plugily/projects/villagedefense/events/Events.java [hunk 3]
```diff
@@ -433,39 +436,53 @@ public void onCraft(CBPlayerInteractEvent event) {
...
-      Arena arena = ArenaRegistry.getArena((Player) entity);
-      if(arena == null) {
-        continue;
-      }
-      arena.addOptionValue(ArenaOption.ROTTEN_FLESH_AMOUNT, e.getItem().getItemStack().getAmount());
-      e.getItem().remove();
-      e.setCancelled(true);
-      e.getInventory().clear();
```
Comment (Tigerpanzer02) [inline src/main/java/plugily/projects/villagedefense/events/Events.java:461]: Why isn't that readded on the new part? (Or did you do tests and it worked well without it?)

Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/events/Events.java:461]: It works without it. We have already cleared it when checking arenas.

**Pull request merged**